packages/fuel-indexer-api-server/src/uses.rs

```diff
@@ -293,6 +293,78 @@ pub(crate) async fn register_index_assets(
     Err(ApiError::default())
 }
 
+pub(crate) async fn get_nonce(
+    Extension(pool): Extension<IndexerConnectionPool>,
+) -> ApiResult<axum::Json<Value>> {
+    let mut conn = pool.acquire().await?;
+    let nonce = queries::create_nonce(&mut conn).await?;
+
+    Ok(Json(json!(nonce)))
+}
+
+pub(crate) async fn verify_signature(
+    Extension(config): Extension<IndexerConfig>,
+    Extension(pool): Extension<IndexerConnectionPool>,
+    Json(payload): Json<VerifySignatureRequest>,
+) -> ApiResult<axum::Json<Value>> {
+    if config.authentication.enabled {
+        let mut conn = pool.acquire().await?;
+        match config.authentication.strategy {
+            Some(AuthenticationStrategy::JWT) => {
+                let nonce = queries::get_nonce(&mut conn, &payload.message).await?;
+                queries::delete_nonce(&mut conn, &nonce).await?;
```

Comment:
Should we wait to delete the nonce until we know that the signature is at least recovered as it can possibly fail? Just a thought; I suppose if the signature doesn't get verified, then a new nonce should be used anyways?